src/platform/nrfconnect/wifi/WiFiManager.cpp

```diff
@@ -188,7 +188,18 @@ CHIP_ERROR WiFiManager::Scan(const ByteSpan & ssid, ScanResultCallback resultCal
     mWiFiState          = WIFI_STATE_SCANNING;
     mSsidFound          = false;
 
-    if (0 != net_mgmt(NET_REQUEST_WIFI_SCAN, mNetIf, NULL, 0))
+    wifi_scan_params * scanParamsFinal{ nullptr };
+
+    if (!ssid.empty())
+    {
+        /* We must assume that the ssid is handled as a NULL-terminated string.
+           Note that the mScanSsidBuffer is initialized with zeros. */
+        VerifyOrReturnError(ssid.size() < sizeof(mScanParams), CHIP_ERROR_INVALID_ARGUMENT);
+        memcpy(mScanSsidBuffer, ssid.data(), ssid.size());
+        mScanParams.ssids[0] = mScanSsidBuffer;
```

Comment:
as there is no length of SSIDs, the driver relies on `ssids[1] = NULL` to know the length.

Reply:
I see, shall we set the max_bss_cnt to 1 in such a case?

Reply:
no, `max_bss_cnt` is for limiting scan results, it's not used for limiting SSIDs